module/skill/GetUp/src/GetUp.cpp

```diff
@@ -60,67 +61,73 @@ namespace module::skill {
+                        if (!body.done) {
+                           // Transform to torso {t} from world {w} space
+                            Eigen::Isometry3d Hwt = sensors.Htw.inverse();
+
+                            // Decompose our basis axes of the torso {t} into world {w} space
+                            Eigen::Vector3d uXTw = Hwt.rotation().col(0);
+                            Eigen::Vector3d uYTw = Hwt.rotation().col(1);
+                            Eigen::Vector3d uZTw = Hwt.rotation().col(2);
+
+                            // Split into six cases (think faces of a cube) to work out which getup script we should run
+
+                            // torso x is mostly world -z
+                            bool on_front = (uXTw.z() <= uXTw.x() && uXTw.z() <= uXTw.y());
+                            // torso x is mostly world z
+                            bool on_back = (uXTw.z() >= uXTw.x() && uXTw.z() >= uXTw.y());
+                            // torso y is mostly world z
+                            bool on_right = (uYTw.z() >= uYTw.x() && uYTw.z() >= uYTw.y());
+                            // torso y is mostly world -z
+                            bool on_left = (uYTw.z() <= uYTw.x() && uYTw.z() <= uYTw.y());
+                            // torso z is mostly world z
+                            bool upright = (uZTw.z() >= uZTw.x() && uZTw.z() >= uZTw.y());
+                            // torso z is mostly world -z
+                            bool upside_down = (uZTw.z() <= uZTw.x() && uZTw.z() <= uZTw.y());
+
+                            if (on_front) {
+                                log<INFO>("Getting up from front");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_front));
+                            }
+                            else if (on_back) {
+                                log<INFO>("Getting up from back");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_back));
+                            }
+                            else if (on_right) {
+                                log<INFO>("Getting up from right");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_right));
+                            }
+                            else if (on_left) {
+                                log<INFO>("Getting up from left");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_left));
+                            }
+                            else if (upright) {
+                                log<INFO>("Getting up from upright");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_upright));
+                            }
+                            else if (upside_down) {
+                                log<INFO>("Getting up from upside_down");
+                                emit<Task>(load_script<BodySequence>(cfg.getup_upside_down));
+                            }
```

Comment:
It might be nicer to move this to a function while using an enum for the orientations and use a switch block format instead. In terms of condensing the logic a bit, would converting into a roll-pitch-yaw configuration allow us to simplify this?

Reply:
maybe, but outside the scope of this PR